src/actions/content.js

```diff
@@ -97,7 +97,143 @@ function* loadContent(action) {
     yield put(hideLoading());
   }
 }
+export const ACTION_EXECUTE = 'ACTION_EXECUTE';
+export const actionExecute = (action, nids) => {
+  return {
+    type: ACTION_EXECUTE,
+    payload: {
+      action,
+      nids,
+    },
+  };
+};
+
+export const KNOWN_ACTIONS = [
+  'entity:delete_action:node',
+  'node_make_sticky_action',
+  'node_make_unsticky_action',
+  'node_promote_action',
+  'entity:publish_action:node',
+  'node_unpromote_action',
+  'entity:unpublish_action:node',
+];
+
+// @todo Find a better name
+// @todo How do we update the store with the new values of the nodes
+//    or the deleted nodes.
+export function* doActionExecute({ payload: { action, nids } }) {
+  try {
+    const contentList = yield select(state => state.content.contentList);
+    const actions = nids
+      .map(nid => {
+        const node = contentList.filter(
+          contentItem => String(contentItem.attributes.nid) === nid,
+        )[0];
+
+        let saveAction;
+        switch (action.attributes.plugin) {
```

Comment:
Instead of doing the API calls directly I'm wondering whether this should trigger NODE_DELETE and NODE_SAVE actions, which then update the store, but also trigger the API calls.❓

Reply:
I implemented this